src/ppl/src/Slots.h

```diff
@@ -95,6 +106,22 @@ struct Section
   int used_slots;
   int num_slots;
   Edge edge;
+
+  int getMaxContiguousSlots(const std::vector<Slot>& slots)
+  {
+    int max_contioguous_slots = std::numeric_limits<int>::min();
+    for (int i = begin_slot; i <= end_slot; i++) {
+      int contiguous_slots = 0;
+      while (i <= end_slot && !slots[i].blocked && !slots[i].used) {
+        contiguous_slots++;
+        i++;
+      }
+
+      max_contioguous_slots = std::max(max_contioguous_slots, contiguous_slots);
```

Comment:
You could advance to the next free slot here.

Reply:
I've decided to move to the next available slot at the beginning of the for loop to avoid missing a slot. Correct me if I'm wrong, but if we do it at the end of the loop, the `i` will be incremented in the for loop, and we will start the next iteration with `i+1`, losing one slot for the contiguous count.